api-specification.md

```diff
@@ -0,0 +1,652 @@
+
+The Flow Results API adheres to the [JSON API](http://jsonapi.org/format/) specification, version 1.0, an open standard for how a client should request that resources be fetched or modified, and how a server should respond to those requests, via JSON.  All API requests and responses defined below reflect the JSON API norms for query parameters and pagination. JSON API also introduces an envelope structure to embed the Flow Results JSON within additional metadata. Adopting JSON API allows API clients to make use of standard [JSON API libraries](http://jsonapi.org/implementations/).
+
+## API Endpoints
+
+Five standard API endpoints are defined for Flow Results servers operating in the Data Aggregator role. Two "push" endpoints are used to send flow results to a system playing the Data Aggregator role.  Three "pull" endpoints are used to access flow results served by a Data Aggregator.
+
+Endpoints are defined relative to a base URL chosen by the implementation, i.e.:
+
+_Base URL_: [https://my.example-flow-results-server/api/v1](https://my.example-flow-results-server/api/v1)
+
+### "Push" endpoints to submit flow results to a Data Aggregator
+
+#### Publish a Package:
+
+This endpoint is used to publish a Package descriptor for the first time to a Data Aggregator.
+
+_URL_: POST \[Base URL\]/flow-results/packages
+
+_Query parameters_: None
+
+_Request body_: The request body shall specify the `type` of `packages`. It shall contain, within the `attributes` parameter, the JSON structure of the Descriptor.
+
+The `id` of the Descriptor can be omitted \(or null\) if the client wants the Data Aggregator \(server\) to assign a new UUID for this package.  If an `id` is provided, the client must ensure that the UUID does not conflict with a Package `id` that already exists on the Data Aggregator; servers must reject requests with `id` conflicts. This is consistent with the [JSON API specification](http://jsonapi.org/format/#crud-creating-client-ids) for client-generated IDs:
+
+A server MAY accept a client-generated ID along with a request to create a resource. An ID MUST be specified with an id key, the value of which MUST be a universally unique identifier. The client SHOULD use a properly generated and formatted UUID as described in RFC 4122 \[RFC4122\]. A server MUST return 403 Forbidden in response to an unsupported request to create a resource with a client-generated ID.
+
+_Request example:_
+
+```
+POST [Base URL]/flow-results/packages HTTP/1.1
+Content-Type: application/vnd.api+json
+Accept: application/vnd.api+json
+
+{  
+   "data":{  
+      "type":"packages",
+      "attributes":{  
+         "profile":"flow-results-package",
+         "name":"standard_test_survey",
+         "flow-results-specification":"1.0.0-rc1",
+         "created":"2015-11-26 02:59:24+00:00",
+         "modified":"2017-12-04 15:54:44+00:00",
+         "id":null,
+         "title":"Standard Test Survey",
+         "resources":[  
+            {  
```

Comment:
Hi! I'm implementing FLOIP for InSTEDD's Verboice (I know, weird place for introductions 😅 ). I think this example is missing a `name` property at the resource item level, according to https://frictionlessdata.io/specs/data-resource/#name. Also, the official Data Package json-schema requires it (that is how I found this issue): https://frictionlessdata.io/schemas/data-package.json